projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts

```diff
@@ -309,10 +309,18 @@ export class FieldsPurger {
    * @returns A new array with the mapped values
    */
   public mapArrayValuesToNull(array: any[]): any[] {
-    return array.map(element => {
-      return typeof element === 'object'
-        ? Object.assign({}, ...Object.keys(element).map(k => ({ [k]: null })))
-        : null;
-    });
+    if (array && array.length > 0) {
+      return array.map(element => {
+        if (element !== undefined || element === null) {
```

Comment:
Not sure what element === null is adding here. Would work the same without that - or am I missing something

Reply:
@andywilkinshmcts sometimes the value returned is NULL and not just undefined. In such situation, the code fails and that's why I added that extra check.